src/webnn_native/ops/Conv2d.cpp

```diff
@@ -91,33 +38,32 @@ namespace webnn_native { namespace op {
     MaybeError Conv2d::CalculateShape() {
         auto inputShape = mInputs[0]->Shape();
         auto filterShape = mInputs[1]->Shape();
-
         bool nchw = mOptions.inputLayout == ml::InputOperandLayout::Nchw;
-        int32_t inputHeight = nchw ? inputShape[2] : inputShape[1];
-        int32_t inputWidth = nchw ? inputShape[3] : inputShape[2];
-        int32_t inputChannels = nchw ? inputShape[1] : inputShape[3];
 
+        int32_t batchSize = 0, inputChannels = 0, inputHeight = 0, inputWidth = 0;
+        utils::ParseInputOperand(mOptions.inputLayout, inputShape, batchSize, inputChannels,
```

Comment:
How about defining it like below:

std::vector<int32_t> GetInputShape(const OperandBase* input, ml::InputOperandLayout inputLayout) {
}
you can call it in L.39 with` auto inputShape = GetInputShape(mInputs[0], mOptions.inputLayout)`, another concern is why do you use the enum not` bool nchw` defined in L.41

Reply:
Done and use the previous method to get input parameters.